src/libpriv/rpmostree-postprocess.c

```diff
@@ -1149,23 +1153,25 @@ rpmostree_copy_additional_files (GFile         *rootfs,
       dest = _rpmostree_jsonutil_array_require_string_element (add_el, 1, error);
       if (!dest)
         return FALSE;
+      dest += strspn (dest, "/");
+      if (!*dest)
+        return glnx_throw (error, "Invalid destination in add-files");
 
-      {
-        g_autoptr(GFile) srcfile = g_file_resolve_relative_path (context_directory, src);
-        const char *rootfs_path = gs_file_get_path_cached (rootfs);
-        g_autofree char *destpath = g_strconcat (rootfs_path, "/", dest, NULL);
-        g_autoptr(GFile) destfile = g_file_resolve_relative_path (targetroot, destpath);
-        g_autoptr(GFile) target_tmpfilesd_parent = g_file_get_parent (destfile);
+      g_print ("Adding file '%s'\n", dest);
 
-        g_print ("Adding file '%s'\n", dest);
+      g_string_truncate (dnbuf, 0);
+      g_string_append (dnbuf, dest);
+      const char *dn = dirname (dnbuf->str);
+      dn += strspn (dest, "/");
```

Comment:
We don't need this second `strspn`, right?

Reply:
But the first `strspn` will have already trimmed it to just `foo.txt`, no? We don't modify it in any way after that and before this, right?

Reply:
Yeah, you're right.  But I dunno...the cost of having it is small, the failure case from writing outside our expected root is ugly.  I'd prefer to be conservative around our fd-relative code in general.

Reply:
Made it into an assertion :arrow_down:

Reply:
Yeah, that makes sense to me. Just wanted to make sure I wasn't missing anything!